ui/components/app/transaction-list/transaction-list.component.js

```diff
@@ -251,6 +353,58 @@ export default function TransactionList({
     return dateGroup;
   };
 
+  const renderFilterButton = () => {
+    return (
+      <Box
+        display={Display.Flex}
+        padding={4}
+        justifyContent={
+          isFullScreen ? JustifyContent.flexStart : JustifyContent.spaceBetween
+        }
+        ref={popoverRef}
+      >
+        <ButtonBase
```

Comment:
This seems like a copy/paste from Nick's PR, which is fine _for now_, but we'll need to make this component more generic afterward so it can be used in both places.

Reply:
I agree with you. There's also other components we could unify and generalize. I left a comment on Nick's PR about it [here](https://github.com/MetaMask/metamask-extension/pull/27884#discussion_r1822788644),  it might be a good enhancement to consider for the future.